### PR TITLE
add MolarEnergy, implement Energy unimplemented method

### PR DESCRIPTION
**Pull request by underscorenico**
This is the 5th PR concerning #252 and the first one implementing unimplemented methods and missing units in energy.

### CODE REVIEW

File: shared/src/main/scala/squants/energy/MolarEnergy.scala [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+import squants.mass.{ChemicalAmount, Moles}
+import squants.{AbstractQuantityNumeric, Dimension, PrimaryUnit, Quantity, SiUnit, UnitConverter, UnitOfMeasure}
+
+/**
+  *
+  * @author Nicolas Vinuesa
+  * @since 1.4
+  *
+  * @param value Double
```
Comment (cquiroz) [inline shared/src/main/scala/squants/energy/MolarEnergy.scala]: Some description?

Comment (cquiroz) [inline shared/src/main/scala/squants/energy/MolarEnergy.scala]: I mean, I'm not very familiar with `MolarEnergy`, perhaps we could add a link to wikipedia?

Comment (underscorenico) [inline shared/src/main/scala/squants/energy/MolarEnergy.scala]: Its just a unit of energy over mass. The link could be: https://en.wikipedia.org/wiki/Joule_per_mole
Should I add it in the class? I've not seen it elsewhere...

Comment (cquiroz) [inline shared/src/main/scala/squants/energy/MolarEnergy.scala]: Hmm, you're right. I've seen only one case but that is in another level. Nevermind

**Comment by derekmorr** [comment]
LGTM. We'll need to add this to `ImplicitDimensions` too.

**Comment by underscorenico** [comment]
@derekmorr it is already in implicitDimensions ... check the commit

**Comment by derekmorr** [comment]
Ah, sorry. Too many commits this morning.

**Comment by garyKeorkunian** [comment]
I think that misspelling of JoulesToMol*e* should be corrected before this is merged.

**Comment by underscorenico** [comment]
Of course, i'll get it done right away

**Comment by underscorenico** [comment]
@garyKeorkunian done!

**Pull request merged**